eventlet/tpool.py

```diff
@@ -333,4 +338,12 @@ def killall():
 
 def set_num_threads(nthreads):
     global _nthreads
-    _nthreads = nthreads
+    old_nthreads, _nthreads = _nthreads, nthreads
+
+    if _setup_already:
+        if old_nthreads > nthreads:
+            import warnings
```

Comment:
import inside function is a test for luck, please never do it again.

Reply:
I'll more than gladly change this, as I only did it like this in the first place because I always try to follow the coding style and patterns used in the project I'm contributing to, even when they don't match my own, and in this case we have this same pattern of importing "warnings" in the "setup" method.